### PR TITLE
Add README.md files to improve documentation UX

### PR DESCRIPTION
**Pull request by ppbaena**
Signed-off-by: Pepe Baena [jbaena@vmware.com](jbaena@vmware.com)

### Description of the change

Once documentation structure has been updated in #4123, next step is to increase readability and findability for users and contributors. This PR aims to add a set of README files for each folder to contribute on this purpose.

### Benefits

Simplifying access to documentation.

### Possible drawbacks

N/A

### Applicable issues

N/A

### Additional information

N/A

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+## Documentation overview
+
+A high-level overview of how it’s organized will help Kubeapps contributors and users know where to look for certain things.
+
+| Section                      | Description                                                                              |
+| ---------------------------- | ---------------------------------------------------------------------------------------- |
+| [Tutorials:](./tutorials/)   | Start here if you’re new to Kubeapps. A hands-on introduction to Kubeapps for new users. |
+| [How-to guides:](./howto/)   | They guide you through the steps involved in addressing key problems and use-cases.      |
+| [Background:](./background/) | Dive into the overall architecture and implementation details of Kubeapps.               |
+| [Reference:](./reference/)   | Technical information - developer guides, design proposals, examples, translations, etc. |
```
Comment (absoludity) [inline docs/README.md:14]: Perhaps remove the colon `:` in the first column, as you've done for the other tables in the sub docs?

File: docs/background/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+| -------------------------------------------------- | ------------------------------------------------------------------------------------------------ |
+| [Architecture](./architecture.md)                  | Dive into the overall Kubeapps architecture.                                                     |
+| [Kubeapps API documentation](./update-api-docs.md) | Information about Kubeapps API and how to contribute to keep Kubeapps API documentation updated. |
+
+Alternatively, our [Tutorials section](../tutorials/README.md) contain step-by-step tutorials to help outline what Kubeapps is capable of while helping you achieve specific aims, such as installing Kubeapps or using an OIDC provider.
+
+If you have a specific goal, but are already familiar with Kubeapps, take a look at our [How-to guides](../howto/README.md). These have more in-depth detail and can be applied to a broader set of features.
+
+Finally, take a look at our [Reference section](../reference/README.md) when you need to know design decisions, what functions the API supports, detailed developer guides, etc.
```
Comment (absoludity) [inline docs/background/README.md:14]: Excellent - I like the way each section still helps you to get to the other sections when relevant.

File: docs/howto/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+| How-to-guides                                                                          | Get stuff done                                                                                                                            |
+| -------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------- |
+| [Access Control](./access-control.md)                                                  | Establish how users will authenticate with the Kubernetes clusters on which Kubeapps operates.                                            |
+| [Basic Form Support](./basic-form-support.md)                                          | Configure your Helm chart in order to present this basic form to the users.                                                               |
```
Comment (absoludity) [inline docs/howto/README.md:8]:
```suggestion
| [Basic Form Support](./basic-form-support.md)                                          | Configure your Helm chart in order to present a simple intuitive form during installation.                                                               |
```

**Review by absoludity — APPROVED** [review]
Excellent Pepe. I really like the way you've linked back to other sections helping educate the reader to find what they need.

File: docs/README.md (hunk 1 above)
Comment (ppbaena) [reply docs/README.md:14]: +1. Thanks for the heads up.

**Pull request merged**